sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc

```diff
@@ -93,7 +93,7 @@ The +-fsycl-core-sub-group-size+ flag compiles all kernels in the translation un
 
 === Sub-group Queries
 
-Several aspects of sub-group functionality are implementation-defined: the size and number of sub-groups for certain work-group sizes is implementation-defined (and may differ for each kernel); and different devices may make different guarantees with respect to how sub-groups within a work-group are scheduled.  Developers can query these behaviors at a device level and for individual kernels.  The sub-group size for a given combination of kernel and launch configuration is fixed, and guaranteed to be reflected by device and kernel queries.
```

Comment:
> The sentence containing this phrase has also been reworked to avoid
> implying that all properties of a sub-group can be queried, since this
> is currently untrue.

Honestly, I wasn't under impression that the last sentence was implying that _all_ properties of a sub-group can be queried, because it specifically mentioned sub-group size and nothing more. Probably I'm missing something

Reply:
The issue here was pointed out by @gmlueck.  It is not always possible to query the sub-group size that is actually used for a given work-group size, because what the query returns is the *max* work-group size (i.e. the maximum sub-group size that can be supported by a compiled kernel).

For example, a work-group size of 7 may return 8 for its `max_work_group_size`, and there is no query that returns 7.